### PR TITLE
ModuleTracker : Implement Archive features

### LINKED ISSUE #470: ModuleTracker : Implement archive features
Implement basic archiving features : 
-ArchiveModule
-UnarchiveModule
-View archived modules

### PR DESCRIPTION
**Pull request by davidcaiqifan**
Fixes #470

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -91,30 +94,71 @@ public void init() throws Exception {
...
             logger.warning("Problem while reading from the file. Will be starting with an empty ModuleList");
-            initialData = new ModuleList();
+            initialModuleList = new ModuleList();
         }
-        return new ModelManager(initialData, initialContactList, initialTodoList, initialEventList, userPrefs);
+        return initialModuleList;
     }
 
+    /**
+     * Returns a {@code ReadOnlyContactList} with the data from {@code storage}'s Archived Module list.
```
Comment (jonasngs) [inline src/main/java/seedu/address/MainApp.java:137]: Maybe can take note of some minor javadocs issues here.

File: src/main/java/seedu/address/logic/commands/modulelistcommands/ViewArchivedModulesCommand.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Deletes a module identified using it's displayed index from the address book.
```
Comment (jonasngs) [inline src/main/java/seedu/address/logic/commands/modulelistcommands/ViewArchivedModulesCommand.java:11]: Javadocs issue here.

File: src/main/java/seedu/address/logic/parser/modulelistparsers/ViewArchivedModulesParser.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package seedu.address.logic.parser.modulelistparsers;
+
+import seedu.address.logic.commands.modulelistcommands.ViewArchivedModulesCommand;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public class ViewArchivedModulesParser implements Parser<ViewArchivedModulesCommand> {
```
Comment (jonasngs) [inline src/main/java/seedu/address/logic/parser/modulelistparsers/ViewArchivedModulesParser.java:7]: If there is no need to parse the user inputs, we can return the command in the ModuleListParser itself, I think this class is redundant.

**Review by jonasngs — CHANGES_REQUESTED** [review]
LGTM, just some minor issues to take note of.

**Review by jonasngs — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/modulelistparsers/ViewArchivedModulesParser.java (hunk 3 above)
Comment (davidcaiqifan) [reply src/main/java/seedu/address/logic/parser/modulelistparsers/ViewArchivedModulesParser.java:7]: Alright noted thanks.

**Pull request merged**